### ISSUE TITLE
Object list is not always correctly ordered

### ISSUE THREAD

**Issue by backwardsEric** [angband/angband#4664]
See Selkie's post here, http://angband.oook.cz/forum/showthread.php?t=10493&page=6 .  The version involved isn't explicitly mentioned but seems likely to be 4.2.1.

A possible cause is this:  compare_items() doesn't provide a total order since with two unknown items, o1 and o2, compare_items(o1, o2) is 1 and compare_items(o2, o1) is also 1.

**Comment by NickMcConnell** [comment]
It doesn't seem to me that that accounts for unknown items coming before known items.

**Comment by backwardsEric** [comment]
I agree that there's likely something else going on.  The handling of the return value when both objects are unknown is the one thing I've spotted that doesn't match what qsort() expects from the comparison function.

I haven't been able to reproduce the problem running the current code on a Mac.

**Comment by NickMcConnell** [comment]
At the risk of having too many Ongoing issues, I'm assigning this to Ongoing...
